src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -51,20 +87,28 @@ object ResourceMatcher {
       badConstraints.isEmpty
     }
 
-    def portsOpt: Option[Seq[RangesResource]] = new PortsMatcher(app, offer, acceptedResourceRoles).portRanges match {
-      case None =>
-        log.warn("App ports are not available in the offer.")
-        None
+    def portsOpt: Option[Seq[RangesResource]] = new PortsMatcher(app, offer, acceptedResourceRoles).portRanges
+
+    val basicResourceMatches = Map(
+      "cpu" -> cpuMatchOpt,
+      "disk" -> diskMatchOpt,
+      "mem" -> memMatchOpt
+    )
 
-      case x @ Some(portRanges) =>
-        log.debug("Met all constraints.")
-        x
+    if (!basicResourceMatches.values.forAll(_.map(_.matches).getOrElse(false))) {
+      val basicResourceString = basicResourceMatches.map {
+        case (resource, Some(scalarMatch)) =>
+          s"$resource $scalarMatch"
+        case (resource, None) =>
+          s"$resource not in offer"
+      }.mkString(", ")
+      log.info(s"Offer ${offer.getId.getValue}. Not all basic resources satisfied: $basicResourceString")
     }
 
     for {
-      cpuRole <- cpuRoleOpt
-      memRole <- memRoleOpt
-      diskRole <- diskRoleOpt
+      cpuRole <- cpuMatchOpt.flatMap(_.matchingRole)
+      memRole <- memMatchOpt.flatMap(_.matchingRole)
+      diskRole <- diskMatchOpt.flatMap(_.matchingRole)
```

Comment:
What do you think about eliminating `ScalarResource.matchingRole` and rewriting this as follows:

```scala
diskMatchOpt.filter(_.matches).flatMap(_.matchingRole)
```